gocd/bci_repo_publish.py

```diff
@@ -95,6 +96,13 @@ def run(self, version, token=None):
                 'publish_prj': f'SUSE:Products:SLE-BCI:{version}:{arch}'
             })
 
+        if not self.is_repo_published(build_prj, 'images'):
+            self.logger.info(f'{build_prj}/images not successfully built')
+            pkg = packages[0]
+            self.last_publish_mtime = \
+                self.mtime_of_product(pkg['publish_prj'], pkg['name'], 'images', 'local')
```

Comment:
and here? we're not actually publishing, we're aborting because it wasn't successfully built?

Reply:
Yes, which means we have to fetch the last published mtime from OBS to know how long ago that was

Reply:
as a nit pick, people usually avoid escaping newlines in python. you can write this in a more pythonic way:

``` python

     self.last_publish_mtime = self.mtime_of_product(
           pkg['publish_prj'], pkg['name'], 'images', 'local')
```

Reply:
IMO harder to read and flake8 didn't complain :shrug: